homeassistant/components/sensor/magicseaweed.py

```diff
@@ -0,0 +1,206 @@
+"""
+Support for magicseaweed data from magicseaweed.com.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.magicseaweed/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, CONF_NAME, CONF_MONITORED_CONDITIONS, ATTR_ATTRIBUTION)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['magicseaweed==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOURS = 'hours'
+CONF_SPOT_ID = 'spot_id'
+CONF_UNITS = 'units'
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+DEFAULT_UNIT = 'us'
+DEFAULT_NAME = 'MSW'
+DEFAULT_ATTRIBUTION = "Data provided by magicseaweed.com"
+
+ICON = 'mdi:waves'
+
+HOURS = ['12AM', '3AM', '6AM', '9AM', '12PM', '3PM', '6PM', '9PM']
+
+SENSOR_TYPES = {
+    'max_breaking_swell': ['Max'],
+    'min_breaking_swell': ['Min'],
+    'swell_forecast': ['Forecast'],
+}
+
+UNITS = ['eu', 'uk', 'us']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SPOT_ID): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOURS, default=None):
+        vol.All(cv.ensure_list, [vol.In(HOURS)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNITS): vol.In(UNITS),
+})
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Magicseaweed sensor."""
+    name = config.get(CONF_NAME)
+    spot_id = config[CONF_SPOT_ID]
+    api_key = config[CONF_API_KEY]
+    hours = config.get(CONF_HOURS)
+
+    if CONF_UNITS in config:
+        units = config.get(CONF_UNITS)
+    elif hass.config.units.is_metric:
+        units = UNITS[0]
+    else:
+        units = UNITS[2]
+
+    forecast_data = MagicSeaweedData(
+        api_key=api_key,
+        spot_id=spot_id,
+        units=units)
+    forecast_data.update()
+
+    # If connection failed don't setup platform.
+    if forecast_data.currently is None or forecast_data.hourly is None:
+        return False
+
+    sensors = []
+    for variable in config[CONF_MONITORED_CONDITIONS]:
+        if 'forecast' in variable:
```

Comment:
This check can be inverted and combined with `if hours is not None` below via `and`.